config/src/config/storage_config.rs

```diff
@@ -113,6 +169,10 @@ pub struct StorageConfig {
     /// since genesis. To recover operation after data loss, or to bootstrap a node in fast sync
     /// mode, the indexer db needs to be copied in from another node.
     pub enable_indexer: bool,
+    /// Fine grained control for db paths of individal databases/shards.
+    /// If not specificed, will use `dir` as default.
+    /// Only allowed when sharding is enabled.
```

Comment:
just curious why sharding enabled is required?

Reply:
Just to reduce the complexity of implementation.. there will be no need to put data on multiple disks if db is not sharded.